py-polars/polars/selectors.py

```diff
@@ -109,8 +111,8 @@ def is_selector(obj: Any) -> bool:
     return isinstance(obj, _selector_proxy_) and hasattr(obj, "_attrs")
 
 
-# TODO: Don't use this. It collects a schema.
-# This should all go to IR conversion.
+# TODO: Don't use this for LazyFrame as it collects a schema.
```

Comment:
We can remove this entirely hopefully as `DataFrame` can dispatch to the same logic as `LazyFrame`. I also think it will be faster to handle on the rust side.

Reply:
No problem, will adjust the comment ;) 
Was doubling as a note to myself to prioritise lazy removal over eager.